src/app/app.ts

```diff
@@ -24,14 +26,14 @@ export interface App {
 }
 
 class AppImpl implements App {
-    call: AppCall
-    context: AppContext
+    call: AppCallRequest
+    context: AppContextWithBot
     values: AppCallValues
 
-    constructor(call: AppCall) {
+    constructor(call: AppCallRequest) {
         this.call = call;
-        this.context = call.context;
-        this.values = call.values;
+        this.context = call.context as AppContextWithBot;
+        this.values = call.values as AppCallValues;
```

Comment:
Is the `as` really needed?

Reply:
@hanzei, yes, `values` field in the `CallRequest` struct is an optional field as seen 

https://github.com/mattermost/mattermost-plugin-apps/blob/master/apps/call.go#L55

and 

https://github.com/mattermost/mattermost-redux/blob/master/src/types/apps.ts#L71